node/src/effect.rs

```diff
@@ -174,26 +174,26 @@ impl<T: 'static + Send> Responder<T> {
     }
 }
 
-impl<T> Responder<T> {
+impl<T> Responder<T>
+where
+    T: Debug,
+{
     /// Send `data` to the origin of the request.
-    pub(crate) fn respond(mut self, data: T) -> impl Future<Output = ()> {
-        let caller_backtrace = backtrace::Backtrace::new();
-        async move {
-            if let Some(sender) = self.0.take() {
-                if sender.send(data).is_err() {
-                    error!(
-                        responder = ?self,
-                        ?caller_backtrace,
-                        "could not send response to request down oneshot channel"
-                    );
-                }
-            } else {
+    pub(crate) async fn respond(mut self, data: T) {
+        if let Some(sender) = self.0.take() {
+            if let Err(data) = sender.send(data) {
                 error!(
                     responder = ?self,
-                    ?caller_backtrace,
-                    "tried to send a value down a responder channel, but it was already used"
+                    ?data,
```

Comment:
If we log the `data` we don't have to log the `responder` as that will be a duplicated information: `self` will log the type of `data` effectively. Same below.

Reply:
Addressed.